packages/xo-server/src/xapi/index.mjs

```diff
@@ -566,8 +566,12 @@ export default class Xapi extends XapiBase {
     for (const vbd of vbds) {
       if (vbd.type === 'Disk') {
         const vdi = vbd.$VDI
-        // Ignore VDI snapshots which have a parent
-        if (vdi.$snapshot_of !== undefined) {
+        // Prevent VDI_NOT_IN_MAP error
+        // The problem comes from XAPI itself (observed on an XCP-ng 8.1 host, might be related to CBT),
+        // where some non-snapshot VDIs had a snapshot_of property not equal to OpaqueRef:NULL as expected
+        // but containing a reference to an existing VDI.
+        // https://team.vates.fr/vates/pl/9m4u5rsr7tfcdroykwq6i6mcmo
```

Comment:
Your comment is ambiguous, it looks like the `if` is there to prevent the error, we need to make it clear that it's the first condition we are talking about here:

```suggestion
        // We need to explicitly test VDI.is_a_snapshot because there is a problem with XAPI itself (observed on
        // an XCP-ng 8.1 host, might be related to CBT), where some non-snapshot VDIs have a snapshot_of
        // property not equal to OpaqueRef:NULL as expected but containing a reference to an existing VDI.
        //
        // https://team.vates.fr/vates/pl/9m4u5rsr7tfcdroykwq6i6mcmo
```

Reply:
Done